app/src/main/java/com/island/island/Activities/ProfileActivity.java

```diff
@@ -133,34 +136,45 @@ public boolean onOptionsItemSelected(MenuItem item)
         return super.onOptionsItemSelected(item);
     }
 
-    public void startNewPostActivity(View view)
-    {
+    public void startNewPostActivity(View view) {
         Intent newPostIntent = new Intent(ProfileActivity.this, NewPostActivity.class);
         startActivity(newPostIntent);
     }
 
-    public void viewProfileImage(View view)
-    {
+    public void viewProfileImage(View view) {
+        Intent intent = new Intent(this, ImageViewerActivity.class);
+
+        intent.putExtra(ImageViewerActivity.IMAGE_VIEW_URI,
+                mProfile.getProfileImageUri().toString());
+        startActivity(intent);
+    }
+
+    public void viewHeaderImage(View view) {
         Intent intent = new Intent(this, ImageViewerActivity.class);
 
-        // TODO: Get profile image uri from database and send string with intent
-        intent.putExtra(ImageViewerActivity.IMAGE_VIEW_URI, "");
+        intent.putExtra(ImageViewerActivity.IMAGE_VIEW_URI,
+                mProfile.getHeaderImageUri().toString());
         startActivity(intent);
     }
 
     private class GetProfileTask extends AsyncTask<Void, Void, Void> {
         protected Void doInBackground(Void... params) {
             if (!Utils.isUser(getApplicationContext(), mProfileUsername)) {
-                Profile profile = MessageLayer.getMostRecentProfile(
+                ProfileWithImageData profileWithImageData = MessageLayer.getMostRecentProfile(
```

Comment:
Yeah I'm mixing the discussion of future architecture ideas into this PR. Maybe that's not the best idea...